x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -330,8 +330,11 @@ static SigningConfiguration buildSigningConfiguration(RealmConfig config) throws
 
     private static List<X509Credential> buildCredential(RealmConfig config, String prefix, Setting.AffixSetting<String> aliasSetting,
                                                         boolean allowMultiple) {
-        final X509KeyPairSettings keyPairSettings = X509KeyPairSettings.withPrefix(prefix, false);
-        final X509KeyManager keyManager = CertParsingUtils.getKeyManager(keyPairSettings, config.settings(), null, config.env());
+        final SslKeyConfig keyConfig = CertParsingUtils.createKeyConfig(config.settings(), prefix, config.env(), false);
+        if (keyConfig.hasKeyMaterial() == false) {
+            return null;
+        }
+        final X509KeyManager keyManager = keyConfig.createKeyManager();
         if (keyManager == null) {
```

Comment:
Nit but it just caught my eye while scrolling. Only `EmptyKeyConfig#createKeyManager` can return `null` and if this is an `EmptyKeyConfig` then we would have returned already because of `if (keyConfig.hasKeyMaterial() == false) {` above

Reply:
I don't think we want every use of `KeyConfig` to know that the case.
I suppose we could document a contract that is `hasKeyMaterial` is false, then `createKeyManager` may not return null, but otherwise I don't think it makes sense for SamlRealm to know about all the possible behaviours of KeyConfig and what the return value from one method might would imply about others.